src/main/java/org/fenixedu/academic/domain/EnrolmentPeriod.java

```diff
@@ -83,6 +83,10 @@ protected void init(final DegreeCurricularPlan degreeCurricularPlan, final Execu
         setEndDateDateTime(endDate);
     }
 
+    public abstract boolean isForCurricularCourses();
```

Comment:
Adding this is an API break. You should provide default implementations for these methods.